POST.md

```diff
@@ -0,0 +1,253 @@
+
+### Test Environment
+
+- **Hardware**: M2 Max MacBook Pro
+- **Kafka Setup**: Three-broker Docker cluster with 3 partitions
+- **Total Messages**: 100,000 messages per test
+- **Libraries Tested**:
+  - `@platformatic/kafka` - Our library across different versions
+  - `kafkajs` - Popular pure JavaScript client
+  - `node-rdkafka` - Native librdkafka bindings
+  - `@confluentinc/kafka-javascript` - Confluent's official client (both KafkaJS and node-rdkafka backends)
+
+### Test Scenarios
+
+**1. Producer (Single Message)**: Measures throughput when sending one message per operation, simulating real-time event streaming applications.
+
+**2. Producer (Batch Messages)**: Measures throughput when sending 100 messages per operation, simulating high-throughput data pipeline scenarios.
+
+**3. Consumer Performance**: Measures message consumption throughput across different consumption patterns (stream-based and event-driven).
+
+## Performance Results: The Numbers
+
+The key point: **we didn't change any benchmark code across versions**. All performance improvements come purely from library optimizations.
+
+Let's compare the results across our initial baseline and the latest version:
+
+### Producer Performance: Single Message
+
+**Initial Baseline Results:**
+
+```
+┌─────────────────────────────────────────────┬─────────┬────────────────┬───────────┐
+│ Library                                     │ Samples │         Result │ Tolerance │
+├─────────────────────────────────────────────┼─────────┼────────────────┼───────────┤
+│ node-rdkafka                                │     100 │  68.30 op/sec  │ ± 67.58 % │
+│ @confluentinc/kafka-javascript (rdkafka)    │     100 │ 220.26 op/sec  │ ±  1.24 % │
+│ @confluentinc/kafka-javascript (KafkaJS)    │     100 │ 250.59 op/sec  │ ±  1.25 % │
+│ KafkaJS                                     │     100 │ 383.82 op/sec  │ ±  3.91 % │
+│ @platformatic/kafka                         │     100 │ 582.59 op/sec  │ ±  3.97 % │
+└─────────────────────────────────────────────┴─────────┴────────────────┴───────────┘
+```
+
+**Current Results (v1.21.0):**
+
+```
+┌─────────────────────────────────────────────┬─────────┬─────────────────┬───────────┐
+│ Library                                     │ Samples │          Result │ Tolerance │
+├─────────────────────────────────────────────┼─────────┼─────────────────┼───────────┤
+│ node-rdkafka                                │  100000 │  17452.32 op/sec│ ± 18.42 % │
+│ @confluentinc/kafka-javascript (KafkaJS)    │  100000 │  20215.32 op/sec│ ± 13.10 % │
+│ @confluentinc/kafka-javascript (rdkafka)    │  100000 │  20771.13 op/sec│ ± 12.69 % │
+│ KafkaJS                                     │  100000 │  61877.92 op/sec│ ±  0.67 % │
+│ @platformatic/kafka                         │  100000 │  95039.18 op/sec│ ±  0.71 % │
+└─────────────────────────────────────────────┴─────────┴─────────────────┴───────────┘
+```
```

Comment:
In the preamble, we said we didn't change a single line of benchmark code, but the numbers of the _other_ libraries are totally different, so I presume we did change something.